js/src/forum/components/Recaptcha.js

```diff
@@ -11,7 +11,28 @@ export default class Recaptcha extends Component {
 
     oncreate(vnode) {
         super.oncreate(vnode);
-        this.attrs.state.render(vnode.dom.querySelector('.g-recaptcha'));
+        
+        new Promise(resolve => {
+            if (app.recaptchaLoaded) {
+                return resolve();
+            }
+
+            app.recaptchaLoaded = true;
+
+            const script = document.createElement('script');
+            script.src = `https://www.recaptcha.net/recaptcha/api.js?hl=${app.translator.locale}&render=explicit`;
+            script.async = true;
+            script.defer = true;
+            script.onload = resolve;
+            document.body.appendChild(script);
+        }).then(() => {
+            const interval = setInterval(() => {
```

Comment:
Why is this interval here? Is it because you're setting `defer` to `true`? 

I'd rather not have this entire logic here - the NPM package `external-load` does a similar thing, and all you need is `load.js('...')` - it's used in fof/geoip (https://github.com/FriendsOfFlarum/geoip/blob/3849e7e13518f3a79ba834f6a37260647fa4ee13/js/src/forum/components/ZipCodeMap.js#L5)

Reply:
without it i get this

```js
Uncaught (in promise) TypeError: grecaptcha.render is not a function
    at RecaptchaState.render (RecaptchaState.js:16)
    at Recaptcha.js:26
```

even with `external-load`

Reply:
Hm, intersting. It's fine then, although not great.

Reply:
after the promise, `window.recaptcha` is still `undefined`. That's why i added the interval.